### ISSUE TITLE
The requests module does not work after freezing a .py file with the --noconsole option.

### ISSUE THREAD

**Issue by ToshiEAB** [pyinstaller/pyinstaller#5742]
Hello,

I've encountered a strange problem -- The requests module does not work after freezing a .py file with the --noconsole option but it works fine without the option. My python program, shortly after it starts, accesses my own website using the requests module (preinstalled in Anaconda). The access succeeds when displaying a console. But it fails when I hide it. I replicated these results several times and so I'm sure the problem has something to do with the --noconsole option. Has anyone encountered this problem and does anyone know how to solve it?

* pyinstaller version: 4.0 (I downloaded it from this repo and manually installed with "python ./pyinstaller-4.0/bootloader/waf all --target-arch=64bit" and "python setup.py install")
* Version of Python: 3.6 (I'm using Anaconda 3)
* Platform:  Windows 10
* Did you also try this on another platform? -- No

Thanks in advance,
toshiEAB

**Comment by rokm** [comment]
Try using PyInstaller 4.2 and `--debug bootloader`; it should give you additonal dialogs with details about the error.

**Comment by ToshiEAB** [comment]
@rokm 
Thanks for your quick reply. I tried to download PyInstaller 4.2 but Chrome said it's dangerous and so I didn't download it -- I know it's not a spam though. So I tied --debug bootloader with PyInstaller 4.0 and did not see any errors. 

I test my Python codes, one by one, and figured out that the issue was not related to the requests module but the os module (actually, the subprocess module inside the os module). In the os module, there is a code:

proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, bufsize=buffering)

I changed this to:

proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, bufsize=buffering, stderr=subprocess.PIPE, stdin=subprocess.DEVNULL)

Then I no longer see the problem associated with the --noconsole option.

Reference: https://cloud6.net/so/python/3425284
